catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,297 @@
+
+    private FederationAdminService federationAdminService;
+
+    private List<RegistryStore> registryStores;
+
+    private Map<String, List<String>> publications = new ConcurrentHashMap<>();
+
+    @Override
+    public <T> List<Action> getActions(T subject) {
+        String registryId = getRegistryId(subject);
+        if (StringUtils.isBlank(registryId)) {
+            return Collections.emptyList();
+        }
+        List<Action> actions = new ArrayList<>();
+        List<String> currentPublications = publications.get(registryId);
+
+        for (RegistryStore registry : registryStores) {
+            //can't publish to yourself
+            if (!registry.isPushAllowed() || StringUtils.isBlank(registry.getRegistryId())
+                    || registryId.equals(registry.getRegistryId())) {
+                continue;
+            }
+
+            CollectionUtils.addIgnoreNull(actions,
+                    getAction(registryId,
+                            registry.getId(),
+                            !currentPublications.contains(registry.getId())));
+        }
+
+        return actions;
+    }
+
+    @Override
+    public String getId() {
+        return providerId;
+    }
+
+    @Override
+    public <T> boolean canHandle(T subject) {
+        return StringUtils.isNotBlank(getRegistryId(subject));
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+
+        Attribute registryIdAttribute = mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+        if (registryIdAttribute == null || StringUtils.isBlank(registryIdAttribute.getValue()
+                .toString())) {
+            return;
+        }
+        String registryId = registryIdAttribute.getValue()
+                .toString();
+
+        Attribute locationAttribute =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        List<String> locations = new ArrayList<>();
+        if (locationAttribute != null) {
+            locations = locationAttribute.getValues()
+                    .stream()
+                    .map(Object::toString)
+                    .collect(Collectors.toList());
+        }
+        if (event.getTopic()
+                .equals(CREATED_TOPIC) || event.getTopic()
+                .equals(UPDATED_TOPIC)) {
+            publications.put(registryId, locations);
+        } else if (event.getTopic()
+                .equals(DELETED_TOPIC)) {
+            publications.remove(registryId);
+        }
+    }
+
+    public void init() throws FederationAdminException, PrivilegedActionException {
+
+        // Change from here
+        // Remove this subject stuff and change to use Security.runAsAdmin() once the PrivilegedException option has been implemented
```

Comment:
Do we already have a ticket for this? If so, we should add the ticket number here and update the ticket to say that this needs to be updated. If we don't have a ticket, we should create one ;-)